src/utils.ts

```diff
@@ -1344,3 +1344,56 @@ export async function fileIsAccessible(fileName: string, mode?: number) {
     return false;
   }
 }
+
+/** @internal */
+export class MovingWindow {
+  /** Index of the next slot to be overwritten */
+  private writeIndex: number;
+  length: number;
+  samples: Float64Array;
```

Comment:
Any reason not to use a `List`?  that would simplify a number of methods in this class

Reply:
The reason behind doing it this way is that since this is a hot path (adding samples will happen every heartbeat for every server), using a circular buffer will likely be faster than creating and removing nodes from the list that then have to be garbage-collected.

Reply:
It's true that we sample per heartbeat, but does that make it a hot code path?  Heartbeats are processed every `heartbeatFrequencyMS`, which defaults to 10 seconds.  So by default, we're only processing rtt samples once per 10 seconds.

Reply:
Fair point, but since it's something that's user-configurable, it would still be good for us make this have a minimal performance impact. And in general, since this doesn't introduce too much extra complexity while reducing the number of objects we have to create and destroy, I think it's still better for us to do it this way.

Reply:
That's fine.  Generally though I don't like to optimize before we have a functional implementation that is benchmarked.  I'd rather implement an unoptimized, simpler solution, let our benchmarks show it's a problem and _then_ optimize than optimize for things that we haven't shown to cause performance issues.